mlir/test/e2e/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Dependencies for the E2E test only
+set(ROCK_E2E_DEPENDS
+  FileCheck count not
+  miopen-opt
+  miopen-gen
+  mlir-miopen-driver
+  mlir-rocm-runner
+  e2e_tests
+)
+
+set(E2E_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(E2E_GEN_SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/generateE2ETest.py")
+set(CONFIG_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+set(LOCAL_CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/lit.local.cfg)
+
+# Command to generate the E2E tests and a local lit config file to control
+# whether the tests are supported or not.
+# This command will be executed when
+#   The local lit config file is missing OR
+#   Any of the *.toml files are changed.
+add_custom_command(OUTPUT ${LOCAL_CONFIG_FILE}
+  COMMAND echo "if not config.enable_rock_e2e_test or config.no_AMD_GPU:" > ${LOCAL_CONFIG_FILE}
```

Comment:
I don't think this is it.
The LOCAL_CONFIG_FILE should be `COPY`ed in from the source directory.

But the other `DEPENDS` are reasonable

We might also have an `add_custom_command` that doesn't have an `OUTPUT`?

Reply:
We can keep the OUTPUT but use `copy` to generate the OUTPUT.

And when only the tests are missing (but `*.toml` and `lit.local.config` are not changed), they will not be re-generated. Do you want an option to force re-generate all tests or it is enough to only depend on the `*.toml` files and `lit.local.cfg`?

Reply:
Good point. If that happens you either force the target to re-run or delete the `lit.local.cfg` in the output to force a regeneration.

Reply:
Unless you don't like that

Reply:
I added a target `ninja re-gen-e2e-tests` to force generate the e2e tests.